### DISCUSSION TITLE
page.apply_redactions is redacting outside of annotations

### DISCUSSION THREAD

**Discussion by chrismahoney25** [pymupdf/PyMuPDF#3444] [Looking for help]
Thank you for the great library. I'm trying to remove sensitive information but when I apply the redaction it's removing extra data across the entire page, including images. This script was working yesterday and now its not. I'm very new to python.

```
import fitz  # Import PyMuPDF

def test_redaction(pdf_path):
    doc = fitz.Document(pdf_path)
    page = doc[0]  # Assuming the content is on the first page
    rect = fitz.Rect(0, 580, 630, 630)
    page.add_redact_annot(rect)
    page._apply_redactions(images=0, graphics=0, text=1)
    output_path = '_redacted.pdf'
    doc.save(output_path)
    doc.close()
    print("Redaction test completed, document saved to:", output_path)


pdf_path = '/Users/chrismahoney/Desktop/SSBC Convenience Plan 2024 - Salon Arte, MA -  Front Desk Consulting with T. Boccia on 04.29.24.pdf'
test_redaction(pdf_path)
```

heres a as screenshot of the annotations:

![image](https://github.com/pymupdf/PyMuPDF/assets/49373206/abc35ae5-28b7-41f0-906a-dc16cdab1ad4)

and here's an image of the outputted pdf:

![image](https://github.com/pymupdf/PyMuPDF/assets/49373206/f6874578-45a2-4b6b-97bd-13f99df2d1f4)

as you can see, it's like the data is still on the document but not visible and the images are gone. If i copy and paste the highlighted info i can see what is suppose to be there. 

I would appreciate any help :)

**Comment by chrismahoney25** [comment]
Update: the screenshots above are from a program calledpandadoc, when I apply my code to a different PDF it works as expected. I'm guessing this is an issued with the PDF structure itself? any ideas how to work around this?

Thanks!

**Comment by JorjMcKie** [comment]
This is probably a duplicate of issue #3376. The fix (in MuPDF) has been developed and will be rolled out with the next version.
